integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/MessageInController.java

```diff
@@ -0,0 +1,193 @@
+package no.difi.meldingsutveksling.nextbest;
+
+import com.google.common.collect.Lists;
+import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
+import no.difi.meldingsutveksling.noarkexchange.MessageException;
+import no.difi.meldingsutveksling.noarkexchange.StatusMessage;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.*;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+@RestController
+public class MessageInController {
+
+    private static final Logger log = LoggerFactory.getLogger(MessageInController.class);
+
+    @Autowired
+    private IncomingConversationResourceRepository repo;
+
+    @Autowired
+    private IntegrasjonspunktProperties props;
+
+    @RequestMapping(value = "/in/messages", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity getIncomingMessages(
+            @RequestParam(value = "messagetypeId", required = false) String messagetypeId,
+            @RequestParam(value = "conversationId", required = false) String conversationId,
+            @RequestParam(value = "senderId", required = false) String senderId) {
+
+        if (!isNullOrEmpty(conversationId)) {
+            Optional<IncomingConversationResource> resource = Optional.ofNullable(repo.findOne(conversationId));
+            if (resource.isPresent()) {
+                return ResponseEntity.ok(resource.get());
+            }
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No conversation with supplied id found.");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "No conversation with supplied id found." 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)